src/main/java/com/mapzen/valhalla/Instruction.java

```diff
@@ -216,16 +217,22 @@ public String getFullInstructionAfterAction(Context context) {
                 DistanceFormatter.format(distanceInMeters, false));
     }
 
-    public String getSimpleInstruction(Context context) {
+    public String getSimpleInstruction(Context context) throws JSONException {
         return context.getString(R.string.simple_instruction, getHumanTurnInstruction(context),
                 getName());
     }
 
     @Override
     public String toString() {
+        String name = "";
+        try {
+            name = getName();
+        } catch (JSONException e) {
+            Log.e("Json exception", "Unable to get name", e);
+        }
         return String.format(Locale.US, "Instruction: (%.5f, %.5f) %s %s LiveDistanceTo: %d",
                 location.getLatitude(), location.getLongitude(), turn,
-                getName(), liveDistanceToNext);
+                name, liveDistanceToNext);
```

Comment:
not sure what the getName used to return in case of an error ... but is blank the best choice here? in other words shouldn't the whole method blow up if this exception occurs?